test_communication/test/test_publisher_subscriber.cpp

```diff
@@ -62,7 +62,8 @@ rclcpp::SubscriptionBase::SharedPtr subscribe(
   received_messages.assign(expected_messages.size(), false);
 
   auto callback =
-    [&expected_messages, &received_messages](const typename T::SharedPtr received_message) -> void
+    [&expected_messages,
+      &received_messages](const typename T::ConstSharedPtr received_message) -> void
```

Comment:
This is weird, maybe:

```suggestion
    [&expected_messages, &received_messages](
      const typename T::ConstSharedPtr received_message
    ) -> void
```

Instead? Same for the other case.

Reply:
Addressed in 0861228 and cf628b8.